### PR TITLE
Adding Pytest Framework and initial Pytest unittests to DataMover

### PR DESCRIPTION
**Pull request by nsarras**

### CODE REVIEW

File: dss_datamover/tests/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+pytest
```
Comment (grandsuri) [inline dss_datamover/tests/requirements.txt:1]: Better to have version numbers like pytest == x.y.z
Sometimes the newer versions or older versions will run into compat issues.
But the downside of this approach is, we need to update this file when we want to move to newer versions after some regression testing.

**Comment by nsarras** [comment]
All System tests are now passing 

image

**Comment by nsarras** [comment]
All System and Unit-tests are passing 

image

File: dss_datamover/tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,182 @@
...
+
+
+class MockSocket():
+    """
+    Dummy Object for an actual socket, should simulate all basic functions of a socket object
+    """
+    # TODO: finish building out MockSocket class
+    def __init__(self):
+        self.mockhost = "192.168.300.144"
+        self.mockport = "4000"
```
Comment (grandsuri) [inline dss_datamover/tests/conftest.py:110]: Hardcoding IPs is not accepted. Someone has to remember to edit these files when they want to validate on their machines.

File: dss_datamover/tests/pytest_config.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+    "config": "/etc/dss/datamover/config.json",
```
Comment (grandsuri) [inline dss_datamover/tests/pytest_config.json:2]: Add socket related IP/Port in this file

File: dss_datamover/utils/utility.py [hunk 4]
```diff
@@ -181,7 +181,7 @@ def remoteExecution(host, username, password="", cmd="", blocking=False):
         return client, stdin, stdout, stderr
 
 
-def get_s3_prefix(logger, nfs_cluster, prefix=None):
+def get_s3_prefix(logger, nfs_cluster, prefix):
```
Comment (grandsuri) [inline dss_datamover/utils/utility.py:184]: what is the need for this change?

File: dss_datamover/tests/conftest.py (hunk 2 above)
Comment (nsarras) [reply dss_datamover/tests/conftest.py:110]: this is a fake ip / port

File: dss_datamover/tests/pytest_config.json (hunk 3 above)
Comment (nsarras) [reply dss_datamover/tests/pytest_config.json:2]: it is not needed, this is a pytest config file, not a datamover config file.

File: dss_datamover/utils/utility.py (hunk 4 above)
Comment (nsarras) [reply dss_datamover/utils/utility.py:184]: this was for the both the SonarCloud complaint as well as a potential bug. There is an underlying function that uses the prefix that is passed in here. Therefore, a prefix must be passed in and cannot be none

Comment (nsarras) [reply dss_datamover/utils/utility.py:184]: I initially thought this was the issue SonarCloud was complaining about 2v3 args, I will change it back

File: dss_datamover/tests/conftest.py (hunk 2 above)
Comment (nsarras) [reply dss_datamover/tests/conftest.py]: replaced with x chars instead

**Review by grandsuri — APPROVED** [review]
Looks good

**Pull request merged**